lldb/tools/lldb-dap/Handler/InitializeRequestHandler.cpp

```diff
@@ -23,8 +23,15 @@ llvm::Expected<InitializeResponse> InitializeRequestHandler::Run(
     const InitializeRequestArguments &arguments) const {
   dap.clientFeatures = arguments.supportedFeatures;
 
-  // Do not source init files until in/out/err are configured.
-  dap.debugger = lldb::SBDebugger::Create(false);
+  // Check if we already have a shared debugger (eg. for GPU processes), otherwise create individual debugger
+  auto shared_debugger = DAPSessionManager::GetInstance().GetSharedDebugger();
```

Comment:
This is where I don't know how to solve the issue of getting the right debugger from our shared instance. The issue is once anyone sets the shared debugger, we will always get one back here. So if we add functionality later that allows us to follow all child processes of a parent process, they will all always share the same debugger...